certora/specs/v1.5/Execute.spec

```diff
@@ -0,0 +1,167 @@
+/* A specification for the exstensible fallback handler */
+
+
+// ---- Methods block ----------------------------------------------------------
+methods {
+
+    // envfree
+    function numSigsSufficient(bytes signatures,uint256 requiredSignatures) external returns (bool) envfree;
+
+    // function _.isValidSignature(bytes32 _hash, bytes _signature) external => DISPATCHER(true);
+
+    function _.checkTransaction(
+        address to,
+        uint256 value,
+        bytes data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        bytes signatures,
+        address msgSender
+    ) external => DISPATCHER(true) ;
+
+    function _.checkAfterExecution(bytes32 hash, bool success) external 
+        => DISPATCHER(true);
+
+    function Executor.execute(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 txGas
+    ) internal returns (bool) => execute_summary();
+
+    function SecuredTokenTransfer.transferToken(address token, address receiver, uint256 amount) internal returns (bool) => NONDET ;
+    function Safe.handlePayment(
+        uint256 gasUsed,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver
+    ) internal returns (uint256) => NONDET ;
+
+}
+
+// ---- Functions and ghosts ---------------------------------------------------
+
+persistent ghost bool execute_called { init_state axiom execute_called == false; }
+
+function execute_summary() returns bool {
+    execute_called = true ;
+    
+    return true ;
```

Comment:
is there no meaning to checking whether the execute function returned false (and did not execute)?

Reply:
It's not obvious to me if there is. I would rather do something analogous to `return NONDET` so that `execute_called` changes to `true` no matter what the return value, just so the summary is more accurate. I don't know how to do this in CVL though if you have any tips? I would just summarize `execute` as `NONDET` if we weren't making the check. Even so, it's not obvious to me that this affects the property.

Reply:
resolved in commit `f2f39c5`; I summarized it to return true or false non-deterministically